bootstrap.py

```diff
@@ -32,4 +29,82 @@ def main(coderoot: str, argv: Sequence[str] | None = None) -> int:
         print(f"repo {args.repo} not supported yet!")
         return 1
 
+    # xcode-select --install will take a while,
+    # and involves elevated permissions with a GUI,
+    # so best to just let the user go through that separately then retrying,
+    # rather than waiting for it.
+    # There is a way to perform a headless install but it's more complex
+    # (refer to how homebrew does it).
+    try:
+        xcode_git = proc.run(("/usr/bin/xcrun", "-f", "git"))
```

Comment:
Nit: Would it make sense to change `xcode_git` to `git_binary` because it looks like it doesn't matter that its xcode git, it just matters that it's a git_binary

Reply:
thinking about this a bit more i'm just gonna use the git on PATH after xcode git is installed (/usr/bin/git will work afterwards)